src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscores.Fixer.cs

```diff
@@ -19,11 +27,67 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
 
+            var node = root.FindNode(context.Span);
+            if (node == null)
+            {
+                return;
+            }
+
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+            if (symbol == null)
+            {
+                return;
+            }
+
+            var newName = GetNewName(symbol.Name);
+            if (newName.Length == 0)
+            {
+                return;
+            }
+
+            // Make sure there is no symbol with the same name already exists.
+            if (!model.LookupSymbols(context.Span.Start, symbol.ContainingType, newName).IsEmpty)
+            {
+                return;
+            }
+
+            string title = MicrosoftCodeQualityAnalyzersResources.IdentifiersShouldNotContainUnderscoresCodeFixTitle;
+            context.RegisterCodeFix(new MyCodeAction(title,
+                                        async ct => await Renamer.RenameSymbolAsync(context.Document.Project.Solution, symbol, newName, null, ct).ConfigureAwait(false),
+                                        equivalenceKey: title),
+                                    context.Diagnostics);
+        }
+
+        protected static string RemoveUnderscores(string name)
+        {
+            var builder = new StringBuilder();
+            bool isPreviousUnderscore = false;
+            foreach (char c in name)
+            {
+                if (c == '_')
+                {
+                    isPreviousUnderscore = true;
+                    continue;
+                }
+                builder.Append(isPreviousUnderscore ? char.ToUpper(c, CultureInfo.InvariantCulture) : c);
+                isPreviousUnderscore = false;
+            }
+            return builder.ToString();
+        }
+
+        protected abstract string GetNewName(string name);
```

Comment:
Please move this abstract method near the top of the file, ideally just below the constructor.